WireUI/Sources/WireIndividualToTeamMigrationUI/Components/Checkbox.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+import WireDesign
+
+struct Checkbox: View {
+    @Binding var isChecked: Bool
+
+    private let title: AttributedString
+
+    init(isChecked: Binding<Bool>, title: AttributedString) {
+        self._isChecked = isChecked
+        self.title = title
+    }
+
+    init(isChecked: Binding<Bool>, title: String) {
+        self._isChecked = isChecked
+        self.title = AttributedString(title)
+    }
+
+    var body: some View {
+        HStack {
+            Button(action: {
+                isChecked.toggle()
+            }, label: {
+                if isChecked {
+                    Image(systemName: "checkmark.square.fill")
```

Comment:
I believe that in general it is better to rely on the ternary operator in `SwiftUI` to ensure that animations are not broken as this ensures that the `view` has the same structural identity (see: https://swiftwithmajid.com/2021/12/09/structural-identity-in-swiftui/). 

In this case I think the following is preferred:

```swift
Image(systemName: isChecked ? "checkmark.square.fill" : "square" : )
  .font(.system(size: 24))
  .backgroundStyle(Color(uiColor: isChecked ? ColorTheme.Checkbox.enabled : ColorTheme.Checkbox.disabled))
```

Reply:
That is disappointing, wrt readability, but it does indeed seem better. Thanks! Quite an interesting read.